openc3-init/plugins/packages/openc3-tool-common/src/components/widgets/ButtonWidget.vue

```diff
@@ -72,19 +72,26 @@ export default {
       const self = this // needed for $emit
       const screen = this.screen
       const api = this.api
+      const tlm = []
       const run_script = this.runScript
-      lines.forEach((line) => {
+      lines.forEach(async (line) => {
         const result = eval(line.trim())
+        console.log(result)
         if (result instanceof Promise) {
-          result
-            .then((success) => {})
-            .catch((err) => {
-              // This text is in top_level.rb HazardousError.to_s
-              if (err.message.includes('is Hazardous')) {
-                this.lastCmd = err.message.split('\n').pop()
-                this.displaySendHazardous = true
-              }
-            })
+          try {
+            let data = await result
+            // eslint-disable-next-line no-console
+            console.log(data)
+            tlm.push(data)
```

Comment:
This works but because it's async it happens _after_ the code that wants to use it. Not sure how to make this line execute synchronously before the next.

Reply:
Was trying to implement button code like the following:
    
    BUTTON 'Start Collect' "api.tlm('INST', 'PARAMS', 'VALUE5', 'RAW');; var type=screen.get_named_widget('COLLECT_TYPE').text(); api.cmd('INST COLLECT with TYPE '+type+', DURATION '+tlm[0])"

However tlm is null when the next line is executed ... eventually it executes and fills with the value.